core/templates/dev/head/pages/exploration_player/SupplementalCardDirective.js

```diff
@@ -62,6 +71,52 @@ oppia.directive('supplementalCard', [
             $scope.helpCardHasContinueButton = false;
           };
 
+          $scope.isCurrentHintUsable = false;
+
+          $scope.allHintsExhausted = false;
+
+          $scope.promiseForHint = null;
+
+          $scope.hintButtonClicks = 0;
+
+          $scope.tooltipText = '';
+
+          $scope.showHint = function() {
+            var hints = $scope.currentInteractionHints;
+            if ($scope.hintButtonClicks < hints.length) {
```

Comment:
Is this hintButtonClicks or numHintsConsumed? The former could be greater than the latter if the button is clicked repeatedly when disabled.